### ISSUE TITLE
Broken test: org.apache.accumulo.test.functional.SplitRecoveryIT

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#2209]
**Test name(s)**
 - org.apache.accumulo.test.functional.SplitRecoveryIT.test

**Describe the failure observed**
```
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.accumulo.test.functional.SplitRecoveryIT.test(SplitRecoveryIT.java:341)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Testing Environment:**
 - Version of this project: 2.1.0-SNAPSHOT
 - First commit known to fail (or current commit): 548b037c5917dc1e693733dc3d89599369d79103
 - Other environment details:
```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /var/lib/jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.8.1
Java version: 11.0.11, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.11.0.9-5.fc34.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.12.15-300.fc34.x86_64", arch: "amd64", family: "unix"
```

**What have you tried already?**
Re-ran test to verify failure

**Additional context**
This is likely caused by #2189, since it only started occurring with that change.

**Comment by DomGarguilo** [comment]
Finding this IT hard to work with but have tracked down the failure to 
https://github.com/apache/accumulo/blob/8a636a3ba91f5dae1d8b09b095178889a7d79c1d/test/src/main/java/org/apache/accumulo/test/functional/SplitRecoveryIT.java#L192-L193
Other than that, no indication yet on what might be causing this.

**Comment by DomGarguilo** [comment]
More specifically it looks like the error is coming from https://github.com/apache/accumulo/blob/cd58dbb0198601ffda5e1162a757648861c8ea14/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java#L203-L206
Which checks that `fetchedCols.contains(ColumnType.LOADED)`

**Comment by Manno15** [comment]
From my testing, I got this IT to pass by reverting the changes made here https://github.com/DomGarguilo/accumulo/blob/0b82f085073a3d5d3de5ebb22244887b345ea860/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java#L193-L199. Which is just removing the Iterators.filter portion. I am not sure why that piece of the code caused things to error though. ConvertRow is what alters LoadedFiles which ties into @DomGarguilo findings mentioned above.

**Comment by DomGarguilo** [comment]
Looking closer at this test it looks like it is failing from a NullPointerException thrown from calling `getLoaded()` on a null `TabletMetadata` object. It is null because the `TabletsMetadata` object returned from the builder in `AmpleImpl.readTablet()` is empty, triggering the NoSuchElementException, returning null. Still looking into why this is happening.

**Comment by keith-turner** [comment]
Tracked down the problem and fixed it in f503f1f.  After [MetadataTableUtil.splitTablet](https://github.com/apache/accumulo/blob/f503f1f6e66fe5657f305e18a65d07852ba035e0/test/src/main/java/org/apache/accumulo/test/functional/SplitRecoveryIT.java#L209) was called `extent` no longer existed in the metadata table.  Needed to use `high` instead.  The extents `extent` and `high` share the same end row in the test, but have different prev end rows. The changes in #2189 to check the prev end row made this difference matter.

**Comment by DomGarguilo** [comment]
Awesome. LGTM
